### PR TITLE
Fixed Passporting - Handle supervisor invitations to any delegate by …

### PR DESCRIPTION
**Pull request by manishagarwalnhs**
…centre email

### JIRA link
https://hee-tis.atlassian.net/browse/TD-261

### Description
Fixed Passporting - Handle supervisor invitations to any delegate by centre email

### Screenshots
https://hee-tis.atlassian.net/browse/TD-261

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/SupervisorService.cs [hunk 1]
```diff
@@ -216,6 +218,34 @@ public SupervisorDelegateDetail GetSupervisorDelegateDetailsById(int supervisorD
...
+                 @"SELECT da.UserID AS DelegateUserID 
+                            FROM Users u
+                            INNER JOIN DelegateAccounts da
+                            ON da.UserID = u.ID
+                            LEFT JOIN UserCentreDetails ucd
+                            ON ucd.UserID = u.ID
+                            WHERE u.PrimaryEmail = @delegateEmail
+                            AND u.Active = 1 
+                            AND da.CentreID = @centreId", new { delegateEmail, centreId });
+
```
Comment (martinfoster-hee) [inline DigitalLearningSolutions.Data/DataServices/SupervisorService.cs:233]: What does the UserCentreDetails join do here? Should you have ucd.Email = @delegateEmail in the where clause too maybe?

Comment (kevwhitt-hee) [reply DigitalLearningSolutions.Data/DataServices/SupervisorService.cs:233]: I agree, I think we need 
`WHERE (u.PrimaryEmail = @delegateEmail OR ucd.Email = @delegateEmail) ...`

File: DigitalLearningSolutions.Data/DataServices/SupervisorService.cs [hunk 2]
```diff
@@ -216,6 +218,34 @@ public SupervisorDelegateDetail GetSupervisorDelegateDetailsById(int supervisorD
                ).FirstOrDefault();
         }
 
+        public int? ValidateDelegate(int centreId, string delegateEmail)
+        {
+            int? delegateUserId = (int?)connection.ExecuteScalar(
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data/DataServices/SupervisorService.cs:223]: If we are using `ExecuteScalar` - can we be certain that the query will only return 1 result?

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
Need to look at comments in the ValidateDelegate query.

**Review by kevwhitt-hee — APPROVED** [review]

**Pull request merged**